presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/operator/aggregation/UnionAggTest.java

```diff
@@ -0,0 +1,449 @@
+          ::::::  ::::::::
+          ::::::::::::::::
+          ::::::::::::::::
+            ::::::::::
+          ::::::::::::::::
+          ::::::::::::::::
+          ::::::  ::::::::
+          ::::::  ::::::::
+     * The test has three different combinations of geometries which all result in the same shape, to assert that they yield the same result.
+     */
+    @Test
+    public void orderIsIndependent()
+    {
+        // Overlap
+        assertAggregatedGeometries(
+                "order does not affect output, first iteration",
+                "POLYGON ((0 0, 2 0, 2 1, 3 1, 3 0, 5 0, 5 2, 4 2, 4 3, 5 3, 5 5, 3 5, 3 4, 2 4, 2 5, 0 5, 0 3, 1 3, 1 2, 0 2, 0 0))",
+                "POLYGON (( 0 0, 0 2, 2 2, 2 0, 0 0 ))",
+                "POLYGON (( 1 1, 1 4, 4 4, 4 1, 1 1 ))",
+                "POLYGON (( 0 3, 0 5, 2 5, 2 3, 0 3 ))",
+                "POLYGON (( 3 0, 3 2, 5 2, 5 0, 3 0 ))",
+                "POLYGON (( 3 3, 3 5, 5 5, 5 3, 3 3 ))");
+
+        // Overlap
+        assertAggregatedGeometries(
+                "order does not affect output, second iteration",
+                "POLYGON ((0 0, 2 0, 2 1, 3 1, 3 0, 5 0, 5 2, 4 2, 4 3, 5 3, 5 5, 3 5, 3 4, 2 4, 2 5, 0 5, 0 3, 1 3, 1 2, 0 2, 0 0))",
+                "POLYGON (( 0 0, 0 2, 2 2, 2 0, 0 0 ))",
+                "POLYGON (( 0 3, 0 5, 2 5, 2 3, 0 3 ))",
+                "POLYGON (( 3 0, 3 2, 5 2, 5 0, 3 0 ))",
+                "POLYGON (( 3 3, 3 5, 5 5, 5 3, 3 3 ))",
+                "POLYGON (( 1 1, 1 4, 4 4, 4 1, 1 1 ))");
+
+        // Overlap
+        assertAggregatedGeometries(
+                "order does not affect output, third iteration",
+                "POLYGON ((0 0, 2 0, 2 1, 3 1, 3 0, 5 0, 5 2, 4 2, 4 3, 5 3, 5 5, 3 5, 3 4, 2 4, 2 5, 0 5, 0 3, 1 3, 1 2, 0 2, 0 0))",
+                "GEOMETRYCOLLECTION ( POLYGON (( 0 0, 0 2, 2 2, 2 0, 0 0 )), POLYGON (( 0 3, 0 5, 2 5, 2 3, 0 3 )) )",
+                "GEOMETRYCOLLECTION ( POLYGON (( 3 0, 3 2, 5 2, 5 0, 3 0 )), POLYGON (( 3 3, 3 5, 5 5, 5 3, 3 3 )) )",
+                "POLYGON (( 1 1, 1 4, 4 4, 4 1, 1 1 ))");
+    }
+
+    private void assertAggregatedGeometries(String message, String expectedWkt, String... geometriesWkt)
+    {
+        // Add a custom equality assertion because the resulting geometry may have its constituent points in a different order
+        AggregationTestUtils.assertAggregationWithAssertion(function, (left, right) -> {
+            OGCGeometry leftGeometry = OGCGeometry.fromText(left.toString());
+            OGCGeometry rightGeometry = OGCGeometry.fromText(right.toString());
+            // Check for equality by getting the difference
+            if (!leftGeometry.difference(rightGeometry).isEmpty() || !rightGeometry.difference(leftGeometry).isEmpty()) {
+                // If there was a difference, assert on the underlying text for ease of troubleshooting
+                assertEquals(left, right, String.format("Error: test case: %s", message));
+            }
```

Comment:
I initially used `OGCGeometry#Equals`, but depending on how one aggregates, it can result in two geometries which are the same shape, but not equal according to the equals method (because the coordinates are in a different order).  `AggregationTestUtils#assertAggregationInternal` creates scenarios where this happens.